parity/rpc_apis.rs

```diff
@@ -152,21 +180,41 @@ fn to_modules(apis: &[Api]) -> BTreeMap<String, String> {
 }
 
 impl ApiSet {
+	/// Retains only APIs in given set.
+	pub fn retain(self, set: Self) -> Self {
+		ApiSet::List(&self.list_apis() & &set.list_apis())
+	}
+
 	pub fn list_apis(&self) -> HashSet<Api> {
-		let mut safe_list = vec![Api::Web3, Api::Net, Api::Eth, Api::Parity, Api::Traces, Api::Rpc]
-			.into_iter().collect();
+		let mut public_list = vec![
+			Api::Web3, Api::Net, Api::Eth, Api::Parity, Api::Rpc,
+		].into_iter().collect();
```

Comment:
yep, that too. but then you have to either have a `HashSet<&Api>` or add a `Clone` impl